include/mbgl/map/annotation.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef MBGL_MAP_ANNOTATIONS
+#define MBGL_MAP_ANNOTATIONS
+
+#include <mbgl/map/map.hpp>
```

Comment:
The underlying problem is the cyclic dependency between `AnnotationManager` and `Map`. It looks like you can break this by removing `Map& map` from `AnnotationManager` and passing the max zoom to `AnnotationManager::getAnnotationsInBounds` and `AnnotationManager::addPointAnnotations`.